source/common/http/http1/codec_impl.cc

```diff
@@ -669,7 +669,8 @@ ClientConnectionImpl::ClientConnectionImpl(Network::Connection& connection, Conn
 
 bool ClientConnectionImpl::cannotHaveBody() {
   if ((!pending_responses_.empty() && pending_responses_.front().head_request_) ||
-      parser_.status_code == 204 || parser_.status_code == 304) {
+      parser_.status_code == 204 || parser_.status_code == 304 ||
+      (parser_.status_code >= 400 && parser_.content_length == 0)) {
```

Comment:
I don't think this is correct. For example, this should apply to 200: https://tools.ietf.org/html/rfc7230#section-3.3. I would need to look into all the code in more detail, but I think 1xx should never have body (though not end_stream), but I think You might just be able to do >= 200 && parser_.content_length == 0.

Reply:
Nevermind, just reread https://tools.ietf.org/html/rfc7230#section-3.3.2. I think you want to keep 204 and 304 as always no body, but then also >= 200 && content_length == 0. If you could read the RFC in this area and check me on this that would be great. :)

Reply:
Ah I see that is what you did. Great, thanks!